src/main/java/com/altenheim/kalender/controller/logicController/ImportController.java

```diff
@@ -14,5 +44,111 @@ public ImportController(ICalendarEntriesModel allEntries, IWebsiteScraperControl
+                cal = new com.calendarfx.model.Calendar(calName);
+
+            }else
+            {
+                cal = new com.calendarfx.model.Calendar();
+            }
+            
+        } catch (IOException | ParserException e) 
+        {
+            e.printStackTrace();
+            return null;
+        }
+
+        for (int i = 0; i < components.size(); i++)         
+        {
+            if(i < 10)
+            {
+                if(!components.get(i).getClass().equals(VEvent.class))
+                    continue;
+            }
+            var start = (DtStart)(components.get(i).getProperties().getProperty("DTSTART"));
+            var end = (DtEnd)((Property) components.get(i).getProperties().getProperty("DTEND"));
+            var startMilli = start.getDate().toInstant().toEpochMilli();
+            var endMilli = end.getDate().toInstant().toEpochMilli();            
+            var entry = createCalendarFXEntryFromMillis(startMilli, endMilli);
+            var summary = ((Property) components.get(i).getProperties().getProperty("SUMMARY")).getValue();
+            entry.setTitle(summary);
+            var locationProp = (Property) components.get(i).getProperties().getProperty("LOCATION");
+            if(locationProp != null)
+                entry.setLocation(locationProp.getValue());
+            cal.addEntry(entry);
+        }
+        System.out.println(cal.findEntries("").get(0).getInterval().toString());
+        return cal;
+    } 
+
+    public void exportFile(com.calendarfx.model.Calendar cal) throws ValidationException, IOException
+    {
+        var entries = cal.findEntries("");
+        var calName = cal.getName();
+        var icsCalendar = new Calendar();
+        icsCalendar.getProperties().add(new ProdId("-//Smart Planner//iCal4j 1.0//DE"));
+        icsCalendar.getProperties().add(Version.VERSION_2_0);
+        icsCalendar.getProperties().add(CalScale.GREGORIAN);
+        icsCalendar.getProperties().add(new XProperty("X-WR-CALNAME", calName));
+        for(int i = 0; i < entries.size(); i++)
+        {
+            var entry = entries.get(i);
+            icsCalendar.getComponents().add(createIcalEntryFromCalFXEntry(entry));
+        }
+        var fout = new FileOutputStream("testExport.ics");
+        var outputter = new CalendarOutputter();
+        outputter.output(icsCalendar, fout);
+        System.out.println("File created in root directory.");
+    }
+
+    private Entry<String> createCalendarFXEntryFromMillis(long start, long end)
+	{
+		var entry = new Entry<String>();
+		var dateStart = LocalDateTime.ofInstant(Instant.ofEpochMilli(start), ZoneId.systemDefault());
+		var dateEnd = LocalDateTime.ofInstant(Instant.ofEpochMilli(end), ZoneId.systemDefault());		
+		entry.changeStartTime(dateStart.toLocalTime());
+		entry.changeStartDate(dateStart.toLocalDate());
+		entry.changeEndTime(dateEnd.toLocalTime());
+		entry.changeEndDate(dateEnd.toLocalDate());
```

Comment:
auslagern 122-125

Reply:
Warum? Die Methode tut die eine Sache und redundant aufgerufen wird es auch nicht. An der Stelle bisschen egal, weil die Entry später alle aus der Factory kommen unddeshlab auch unten die entry create methode rausfliegen wird. Sind ja copy paste zum testen aus der Suchlogik. Aber begründe mal solche Aussagen in nem Review, das ist wichtig finde ich.